feat: mark deprecated properties with @Deprecated

maltesmann opened the pull request:
<!--   Thank you for your contribution. Before you submit the pull request:
1. Follow our contribution guidelines
2. Test your changes and attach their results to the pull request.
3. Update the relevant documentation.
-->

**Description**

See #418 for details.

**Related issue(s)**
<!-- If you refer to a particular issue, provide its number, otherwise, remove this section.
For example, `Resolves #123`, `Fixes #43`, or `See also #33`. The `See also #33` option will not automatically close the issue after the PR merge. -->
Resolves #418

Tenischev commented:
Hi @maltesmann 
Please update one of tests to cover this functionality. I think [additional-formats.test.js](https://github.com/asyncapi/java-spring-template/blob/3a7f6de220d609494a3ae39c7e79f553b55c428a/tests/additional-formats.test.js) could be the place, but of course you could create a new file.

maltesmann commented:
Hi @Tenischev 
Sure! I just added a deprecated parameter to the test you suggested. I also fixed the jinjer2 whitespace control, to not break the existing tests.

Tenischev commented:
@all-contributors please add @maltesmann for code
